### PR TITLE
Fix finding the end of the fhinal chunk in the presence of incomplete parallel branches [JENKINS-38536]

### PR DESCRIPTION
**Pull request by svanoort**
Fixes [JENKINS-38536](https://issues.jenkins-ci.org/browse/JENKINS-38536) in combination with the work in https://github.com/jenkinsci/workflow-api-plugin/pull/33

### CODE REVIEW

**Comment by svanoort** [comment]
WIP while I sort out why the tests are failing.

**Comment by svanoort** [comment]
@reviewbybees please?

File: rest-api/pom.xml [hunk 1]
```diff
@@ -47,7 +47,8 @@
         <dependency>
             <groupId>org.jenkins-ci.plugins.workflow</groupId>
             <artifactId>workflow-api</artifactId>
-            <version>2.4</version> <!-- allows consuming the new APIs -->
+            <!-- Temp version until final is released -->
+            <version>2.12-alpha</version>
```
Comment (jglick) [inline rest-api/pom.xml]: Be super careful with `alpha`/`beta` dependencies. `maven-release-plugin` will let you go ahead and release this plugin as is—but if anyone actually tries to install it from the non-experimental update center, Jenkins will happily let you go ahead, and then issue an error when you restart. Oops.

So I tend to say that you should only use (timestamped) `SNAPSHOT` dependencies until there is a full upstream release.

**Comment by svanoort** [comment]
@reviewbybees  done

**Pull request merged**